core/src/test/scala/kanaloa/reactive/dispatcher/queue/WorkerWorkingSpec.scala

```diff
@@ -58,32 +57,92 @@ class WorkerWorkingSpec extends WorkerSpec {
       queueProbe.expectMsg(RequestWork(worker)) //asks for more Work now because it is idle
       assertWorkerStatus(worker, Worker.Idle)
     }
 
-    "fail Work, transitions to 'unregisteringIdle' if Terminated(routee)" in withWorkingWorker() { (worker, queueProbe, routeeProbe, work) ⇒
+    "report timeout work to metricsCollector'" in withWorkingWorker(WorkSettings(timeout = 5.milliseconds)) { (worker, _, routeeProbe, _, metricCollectorProbe) ⇒
+      expectMsgType[WorkTimedOut]
+      metricCollectorProbe.expectMsg(Metric.WorkTimedOut)
+    }
+
+    "fail Work, transitions to 'unregisteringIdle' if Terminated(routee)" in withWorkingWorker() { (worker, queueProbe, routeeProbe, work, _) ⇒
       routeeProbe.ref ! PoisonPill
       expectMsgType[WorkFailed]
       queueProbe.expectMsg(Unregister(worker))
       assertWorkerStatus(worker, UnregisteringIdle)
     }
 
-    "transition to 'waitingToTerminate' if Terminated(queue)" in withWorkingWorker() { (worker, queueProbe, routeeProbe, work) ⇒
+    "transition to 'waitingToTerminate' if Terminated(queue)" in withWorkingWorker() { (worker, queueProbe, _, work, _) ⇒
       queueProbe.ref ! PoisonPill
       assertWorkerStatus(worker, Worker.WaitingToTerminate)
     }
 
-    "transition to 'waitingToTerminate' if NoWorkLeft'" in withWorkingWorker() { (worker, queueProbe, routeeProbe, work) ⇒
+    "transition to 'waitingToTerminate' if NoWorkLeft'" in withWorkingWorker() { (worker, _, _, work, _) ⇒
       worker ! NoWorkLeft
       assertWorkerStatus(worker, Worker.WaitingToTerminate)
     }
 
-    "transition to 'unregisteringBusy' if Retire'" in withWorkingWorker() { (worker, queueProbe, routeeProbe, work) ⇒
+    "transition to 'unregisteringBusy' if Retire'" in withWorkingWorker() { (worker, _, _, work, _) ⇒
       worker ! Worker.Retire
       assertWorkerStatus(worker, Worker.UnregisteringBusy)
     }
   }
+
+  "A Working Worker with circuit breaker" should {
+    val cbs = Some(CircuitBreakerSettings(openDurationBase = 50.milliseconds, timeoutCountThreshold = 2))
+    val ws = WorkSettings(timeout = 5.milliseconds)
+    "not trigger circuit breaker when counter is below threshold" in withWorkingWorker(ws, cbs) {
+      (worker, queueProbe, _, work, _) ⇒
+        //after one time out,
+        queueProbe.expectMsg(40.milliseconds, RequestWork(worker))
+
+        worker.underlyingActor.delayBeforeNextWork should be(empty)
+
+    }
+
+    "triggers circuit breaker when counter above threshold, report metrics" in withWorkingWorker(ws, cbs) { (worker, queueProbe, routeeProbe, work, metricCollectorProbe) ⇒
+      queueProbe.expectMsg(RequestWork(worker))
+      queueProbe.send(worker, work) //second time out
+
+      metricCollectorProbe.expectMsg(Metric.WorkTimedOut)
+      metricCollectorProbe.expectMsg(Metric.CircuitBreakerOpened)
+      queueProbe.expectNoMsg((cbs.get.openDurationBase * 2 * 0.9).asInstanceOf[FiniteDuration]) //should not get request for work within delay
```

Comment:
I've run this test many times on travis now (this PR has been synced many times and this test have been fine) travis can only make things slow, i.e. you won't get the message sooner than you expected.